deepcell/datasets/builders.py

```diff
@@ -0,0 +1,209 @@
+        self.morphologies = np.concatenate(morphologies, axis=0)
+        self.adj_matrices = np.concatenate(adj_matrices, axis=0)
+        self.norm_adj_matrices = np.concatenate(norm_adj_matrices, axis=0)
+        self.temporal_adj_matrices = np.concatenate(temporal_adj_matrices, axis=0)
+
+    def _sample_time(self, *args):
+        X_dict = args[0]
+        y_dict = args[1]
+
+        app = X_dict['appearances']
+        max_time = tf.shape(app)[1] - self.track_length
+        t_start = tf.random.uniform([1],
+                                    minval=0,
+                                    maxval=max_time,
+                                    dtype=tf.int32)[0]
+
+        for key in X_dict.keys():
+            data = X_dict[key]
+            if 'adj' not in key:
+                sampled_data = data[:, t_start:t_start + self.track_length, ...]
+                X_dict[key] = sampled_data
+            else:
+                sampled_data = data[:, :, t_start:t_start + self.track_length]
+                X_dict[key] = sampled_data
+
+        for key in y_dict.keys():
+            data = y_dict[key]
+            sampled_data = data[:, :, t_start:t_start + self.track_length - 1, ...]
+            y_dict[key] = sampled_data
+
+        return (X_dict, y_dict)
+
+    def _augment(self, *args):
+        X_dict = args[0]
+        y_dict = args[1]
+
+        app = X_dict['appearances']
+        centroids = X_dict['centroids']
+
+        # Randomly rotate appearances
+        theta = tf.random.uniform([1], 0, 2 * 3.1415926)
+
+        # Transform appearances
+        old_shape = tf.shape(app)
+        new_shape = [-1, tf.shape(app)[2], tf.shape(app)[3], tf.shape(app)[4]]
+        img = tf.reshape(app, new_shape)
+        img = tfa.image.rotate(img, theta)
+        img = tf.reshape(img, old_shape)
+
+        X_dict['appearances'] = img
+
+        # Transform coordinates
+        cos_theta = tf.math.cos(theta)
+        sin_theta = tf.math.sin(theta)
+
+        rot_row_0 = tf.stack([cos_theta, -sin_theta], axis=1)
+        rot_row_1 = tf.stack([sin_theta, cos_theta], axis=1)
+        rotation_matrix = tf.concat([rot_row_0, rot_row_1], axis=0)
+        transformed_centroids = tf.matmul(centroids, tf.transpose(rotation_matrix))
+        r0 = tf.random.uniform([1, 1, 2], -512, 512)
+        transformed_centroids = transformed_centroids + r0
+        X_dict['centroids'] = transformed_centroids
+
+        return (X_dict, y_dict)
+
+    def _create_dataset(self):
+        input_dict = {'appearances': self.appearances,
+                      'centroids': self.centroids,
+                      'morphologies': self.morphologies,
+                      'adj_matrices': self.norm_adj_matrices}
+        output_dict = {'temporal_adj_matrices': self.temporal_adj_matrices}
+        dataset = tf.data.Dataset.from_tensor_slices((input_dict, output_dict))
+        self.dataset = dataset.shuffle(256).repeat().map(self._sample_time).map(self._augment).batch(self.batch_size)
```

Comment:
It looks like all this class does is create this `dataset` value. I'm wondering if it even needs to be a class if that is the case.